exp/api/v1beta1/azuremanagedmachinepool_types.go

```diff
@@ -189,6 +189,11 @@ type AzureManagedMachinePoolSpec struct {
 	// KubeletConfig specifies the kubelet configurations for nodes.
 	// +optional
 	KubeletConfig *KubeletConfig `json:"kubeletConfig,omitempty"`
+
+	// KubeletDiskType specifies the kubelet disk type. Default to OS. Possible values include: 'OS', 'Temporary'
+	// +kubebuilder:validation:Enum=OS;Temporary
+	// +optional
+	KubeletDiskType *string `json:"kubeletDiskType,omitempty"`
```

Comment:
I tried doing this originally, but I was unable to get the typing to work for the agentpool spec [here](https://github.com/willie-yao/cluster-api-provider-azure/blob/8b6f01a615a639df20644ffd0928d778b5e9fc9e/azure/scope/managedmachinepool.go#L182). The agent pool type [definition](https://github.com/willie-yao/cluster-api-provider-azure/blob/8b6f01a615a639df20644ffd0928d778b5e9fc9e/azure/services/agentpools/spec.go#L134) for the KubeletDiskType is unable to reference the type if I declare it as a constant inside of AzureManagedMachinePools, and vice versa.

Reply:
What exactly did you try where you 
I think it would work to leave the field in `AgentPoolSpec` as a `*string` (since that's what the Azure API expects) and then we can convert from the custom type with something like:
```go
		KubeletDiskType:      (*string)(managedMachinePool.Spec.KubeletDiskType),
```

Reply:
I was able to fix this by converting the custom type within the toString method [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2938/commits/c5119e1177d66b951be23793bcb58cc6ec74a66c#diff-2c4585415b3fabfd1913f95eb20bc57daac00c2facd3ffa45d3bb00bb1d41b63L284)! Thanks so much for the suggestion.